### PR TITLE
Update to Valibot 0.31

### PR DESCRIPTION
**Pull request by matt45400**
https://valibot.dev/blog/valibot-v0.31.0-is-finally-available/

### CODE REVIEW

File: packages/valibot/src/validation.ts [hunk 1]
```diff
@@ -21,12 +22,14 @@ export const validationAdapter: ValidationAdapter<
...
-        path: path?.map(({type, key}) =>
-          type === 'map' || type === 'unknown' ? String(key) : key,
-        ),
-      })),
+      issues: result.issues.map((issue) => {
+        const path = getDotPath(issue);
+
+        return {
+          message: issue.message,
+          path: path?.split('.'),
```
Comment (matt45400) [inline packages/valibot/src/validation.ts:30]: Not sure if the path is correct

**Comment by matt45400** [comment]
- [ ] [Link](https://github.com/matt45400/typeschema/blob/9a8fb15151dd6d116dc6e79dbeb0b409705092c3/packages/valibot/src/validation.ts#L20 ) : Types of property 'data' are incompatible. Type 'InferOutput<TSchema>' is not assignable to type 'UnknownIfNever<OutputFrom<AdapterResolver, TSchema>>'.
- [ ] [Link](https://github.com/matt45400/typeschema/blob/9a8fb15151dd6d116dc6e79dbeb0b409705092c3/packages/main/src/__tests__/valibot.test.ts#L50-L51) : Expected 0 arguments, but got 1.

File: packages/valibot/src/validation.ts [hunk 2]
```diff
@@ -4,28 +4,27 @@ import type {ValidationAdapter} from '@typeschema/core';
...
 > = async schema => {
-  const {safeParseAsync} = await importValidationModule();
+  const {getDotPath, safeParseAsync} = await importValidationModule();
   return async data => {
     const result = await safeParseAsync(schema, data);
     if (result.success) {
       return {
-        data: result.output,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        data: result.output as any,
```
Comment (fabian-hiller) [inline packages/valibot/src/validation.ts:20]: @decs we were not able to resolve this type problem. Can you have a look at it?

**Review by decs — APPROVED** [review]
Thanks for making the changes!

**Pull request merged**